test/local_test_c10d_ddp.py

```diff
@@ -0,0 +1,188 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+import argparse
+import os
+import unittest
+
+import pippy
+
+import torch
+import torch.distributed as dist
+import torch.optim as optim
+from pippy.IR import pipe_split, TrivialLossWrapper
+from torch.nn.parallel import DistributedDataParallel
+
+
+d_hid = 512
+chunk_size = 256
+
+torch.manual_seed(0)
+
+
+class ExampleCode(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.mm_param0 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
+        self.mm_param1 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
+        self.mm_param2 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
+        self.lin0 = torch.nn.Linear(d_hid, d_hid)
+        self.lin1 = torch.nn.Linear(d_hid, d_hid)
+
+    def forward(self, x):
+        x = torch.mm(x, self.mm_param0)
+        skip_connection = x
+        x = torch.relu(x)
+        pipe_split()
+        x = torch.mm(x, self.mm_param1)
+        x = self.lin0(x)
+        pipe_split()
+        x = torch.relu(x)
+        x = x + skip_connection
+        x = torch.mm(x, self.mm_param2)
+        pipe_split()
+        x = self.lin1(x)
+        x = torch.relu(x)
+        return x
+
+
+def get_dp_subgroup(args):
+    my_pp_rank = args.rank // args.dp_group_size
+    my_dp_rank = args.rank % args.dp_group_size
+    for pp_rank in range(0, args.pp_group_size):
+        dp_group_ranks = list(
+            range(
+                pp_rank * args.dp_group_size, (pp_rank + 1) * args.dp_group_size
+            )
+        )
+        dp_group = dist.new_group(ranks=dp_group_ranks)
+        if pp_rank == my_pp_rank:
+            my_dp_group = dp_group
+    print(f"Rank {args.rank} done getting dp group")
+    return my_dp_group, my_dp_rank
+
+
+def get_pp_subgroup(args):
+    my_pp_rank = args.rank // args.dp_group_size
+    my_dp_rank = args.rank % args.dp_group_size
+    for dp_rank in range(0, args.dp_group_size):
+        pp_group_ranks = list(
+            range(dp_rank, args.world_size, args.dp_group_size)
+        )
+        pp_group = dist.new_group(ranks=pp_group_ranks)
+        if dp_rank == my_dp_rank:
+            my_pp_group = pp_group
+    print(f"Rank {args.rank} done getting pp group")
+    return my_pp_group, my_pp_rank
+
+
+def run_worker(args):
```

Comment:
Is it possible that we can compare the runs of two models, one is only using DDP and another one is DDP+PP?

Reply:
Added